// engine/map.js
/**
 * 
 */
/* globals MAP_ENGINE */

module.exports = (function () {
	return {
		getCoords : getCoords,
		getLocalX : getLocalX,
		getLocalY : getLocalY,
		getLevel : getLevel,
		createDynamicSquare : createDynamicSquare,
		getDynamicSquare : getDynamicSquare,
		setZone : setZone,
		rotateZone : rotateZone,
		build : build
	};

	function getCoords (node) {
		return MAP_ENGINE.getCoords(node);
	}

	function getLocalX (coord) {
		return MAP_ENGINE.getLocalX(coord);
	}

	function getLocalY (coord) {
		return MAP_ENGINE.getLocalY(coord);
	}

	function getLevel (coord) {
		return MAP_ENGINE.getLevel(coord);
	}

	function createDynamicSquare () {
		return MAP_ENGINE.createArea();
	}

	function getDynamicSquare (coord) {
		return MAP_ENGINE.getDynamicSquare(coord);
	}

	function setZone (mapSquare, level, zoneX, zoneY, srcCoord, rotation) {
		MAP_ENGINE.setZone(mapSquare, zoneX, zoneY, level, srcCoord, rotation);
	}

	function rotateZone (mapSquare, level, zoneX, zoneY, rotation) {
		MAP_ENGINE.rotateZone(mapSquare, zoneX, zoneY, level, rotation);
	}

	function build (mapSquare) {
		MAP_ENGINE.buildArea(mapSquare);
	}
})();
